client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -878,6 +903,23 @@ public int pushOrMergeData(
     // increment batchId
     final int nextBatchId = pushState.nextBatchId();
 
+    if (ioCryptoKey.isPresent()) {
+      CryptoCipher encipher = encipherThreadLocal.get();
+      byte[] encryptData = new byte[length + encipher.getBlockSize()];
+      int encryptLength = CryptoUtils.encrypt(encipher, data, offset, length, encryptData);
```

Comment:
QQ: given we are doing update and final in `encrypt`, do we need to set the size to `length + blockSize * 2` ?
I am blanking out on what the expectation is - please check.